### PR TITLE
chore: Enable automatic API doc generation

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

This change will enable automatic generation of our API docs as part of our release flow. 

Follow up actions once merged:
- Update release MCM template
- Switch API doc tracking branch to `release` to `main`

#### Description of how you validated changes
Verified that the updated steps were being executed correctly in CircleCI against this feature branch: https://app.circleci.com/pipelines/github/aws-amplify/amplify-js/19083/workflows/42186cf0-662e-4729-9155-04ee416974ce/jobs/264363

Generated doc commit (since reverted): https://github.com/aws-amplify/amplify-js/commit/3122c24ce47df50d626864e6d65d5694ce0a9f14

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by AllanZhengYP — APPROVED** [review]
A very handy ops improvement!

File: .circleci/config.yml [hunk 1]
```diff
@@ -1602,6 +1602,17 @@ jobs:
...
+      - run:
+          name: 'Update API documentation'
+          command: |
+            git config --global user.email $GITHUB_EMAIL
+            git config --global user.name $GITHUB_USER
+            git pull origin release
+            yarn run docs
+            git add ./docs/api/
+            git commit -m "chore(release): update API docs [ci skip]"
+            git push origin release
```
Comment (elorzafe) [inline .circleci/config.yml:1614]: I think this should run before Update library version, otherwise `main` and `release` won't be in sync

Comment (jimblanc) [reply .circleci/config.yml:1614]: Good call, putting PR back into draft to test this approach.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1592,17 +1592,50 @@ jobs:
...
+            git add ./packages/core/src/Platform/version.ts
+            git commit -m "chore(release): update version.ts [ci skip]"
+      - run:
+          name: 'Update API documentation'
+          command: |
+            yarn run docs
+            git add ./docs/api/
+            git commit -m "chore(release): update API docs [ci skip]"
+            git push origin chore/docs-generation
+            git push --force-with-lease origin chore/docs-generation
```
Comment (elorzafe) [inline .circleci/config.yml:1637]: what if this is another step?, just to make it clear what is doing each step?

File: .circleci/config.yml (hunk 1 above)
Comment (jimblanc) [reply .circleci/config.yml]: Confirmed that this approach will work: https://app.circleci.com/pipelines/github/aws-amplify/amplify-js/19083/workflows/42186cf0-662e-4729-9155-04ee416974ce/jobs/264363

**Review by elorzafe — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]

**Pull request merged**